libtelemetry/src/main/java/com/mapbox/android/telemetry/AppUserTurnstile.java

```diff
@@ -28,10 +28,10 @@
   @SerializedName("operatingSystem")
   private String operatingSystem = null;
 
-  public AppUserTurnstile(boolean enabledTelemetry, String sdkIdentifier, String sdkVersion, String userId) {
+  public AppUserTurnstile(boolean enabledTelemetry, String sdkIdentifier, String sdkVersion) {
```

Comment:
If `MapboxTelemetry.applicationContext` is `null` when creating an `AppUserTurnstile` event we should throw an `IllegalStateException` to avoid potential `NullPointerException`s when calling `TelemetryUtils.retrieveVendorId()`.

Same idea as in `MapEventFactory` 👀 
https://github.com/mapbox/mapbox-events-android/blob/422c54d2d89ea4bda3686640d3204bc6f9696cb0/libtelemetry/src/main/java/com/mapbox/android/telemetry/MapEventFactory.java#L91-L93